File: hooks/reload-ahk.js
const path = require("path");
const { execSync } = require("child_process");

const AHK_DIR = "C:\\Users\\0000250059\\Documents\\AutoHotkey";
const AHK_EXE = path.join(AHK_DIR, ".tools", "AutoHotkey-v2", "AutoHotkey64.exe");

let raw = "";
process.stdin.on("data", (c) => (raw += c));
process.stdin.on("end", () => {
  const input = JSON.parse(raw);
  const filePath = (input.tool_input || {}).file_path || (input.tool_input || {}).path || "";

  if (!filePath.endsWith(".ahk")) return;

  // Kill all AHK processes
  try {
    execSync("taskkill /F /IM AutoHotkey64.exe", { stdio: "ignore" });
  } catch {}

  // Restart all launcher scripts
  const fs = require("fs");
  const files = fs.readdirSync(AHK_DIR).filter((f) => f.endsWith("-launcher.ahk"));
  for (const f of files) {
    const script = path.join(AHK_DIR, f);
    execSync(`start "" "${AHK_EXE}" "${script}"`, { stdio: "ignore", shell: true });
  }
});

File: hooks/reload-ahk.py
#!/usr/bin/env python3
"""PostToolUse hook (Edit|Write): reload AutoHotkey after .ahk file edits."""

import json
import subprocess
import sys
from pathlib import Path

AHK_DIR = Path.home() / "Documents" / "AutoHotkey"
AHK_EXE = AHK_DIR / ".tools" / "AutoHotkey-v2" / "AutoHotkey64.exe"


def main() -> None:
    data = json.load(sys.stdin)
    tool_input = data.get("tool_input", {})
    file_path = tool_input.get("file_path") or tool_input.get("path", "")

    if not file_path.endswith(".ahk"):
        return

    # Kill all AHK processes
    try:
        subprocess.run(
            ["taskkill", "/F", "/IM", "AutoHotkey64.exe"],
            capture_output=True,
        )
    except FileNotFoundError:
        pass

    # Restart all launcher scripts
    try:
        launchers = [f for f in AHK_DIR.iterdir() if f.name.endswith("-launcher.ahk")]
    except OSError:
        return

    for script in launchers:
        try:
            subprocess.Popen(
                [str(AHK_EXE), str(script)],
                stdout=subprocess.DEVNULL,
                stderr=subprocess.DEVNULL,
                start_new_session=True,
            )
        except FileNotFoundError:
            pass


if __name__ == "__main__":
    main()
